arch/arc/core/arc_smp.c

```diff
@@ -66,17 +66,40 @@ void arch_start_cpu(int cpu_num, k_thread_stack_t *stack, int sz,
 	}
 }
 
+#ifdef CONFIG_SMP
+static void arc_connect_debug_setup(int cpu_num)
```

Comment:
Just a nitpick - probably name it something like `arc_connect_debug_mask_update` so something similar as we call it multiple times.